joint_trajectory_controller/src/trajectory.cpp

```diff
@@ -270,6 +293,68 @@ void Trajectory::interpolate_between_points(
                                 t[2] * 12.0 * coefficients[4] + t[3] * 20.0 * coefficients[5];
     }
   }
+  // Apply velocity, acceleration, and jerk limits with Ruckig
+  if ((period.seconds() > 0) && do_ruckig_smoothing)
+  {
+    // TODO(andyz): do not initialize a new object every cycle, when it's possible.
+    // See https://github.com/pantor/ruckig/issues/118
+    ruckig::Ruckig<ruckig::DynamicDOFs> smoother(dim, period.seconds());
+
+    // If Ruckig has run previously on this trajectory, use the output as input for the next cycle
+    if (have_previous_ruckig_output_)
+    {
+      ruckig_input_.current_position = ruckig_output_.new_position;
+      ruckig_input_.current_velocity = ruckig_output_.new_velocity;
+      ruckig_input_.current_acceleration = ruckig_output_.new_acceleration;
+    }
+    // else, need to initialize the robot state and joint limits
+    else
+    {
+      // For velocity/acceleration/jerk-limited command smoothing
+      ruckig_input_ = ruckig::InputParameter<ruckig::DynamicDOFs>(dim);
+      ruckig_output_ = ruckig::OutputParameter<ruckig::DynamicDOFs>(dim);
```

Comment:
This could also be defined “statically” in the constructor.

Reply:
I don't think they can go in the constructor in case the input `std::shared_ptr<trajectory_msgs::msg::JointTrajectory>` is null. I'll look for a better place to initialize them, though.